File: app/src/main/java/com/joyy/android_project/MainActivity.java
package com.joyy.android_project;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;

public class MainActivity extends BaseActivity2 {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    void test() {
        Handler handler = new Handler();
        handler.sendEmptyMessage(0);
        handler.post(new Runnable() {
            @Override
            public void run() {

            }
        });
    }
}

File: app/src/main/java/com/joyy/android_project/MainActivity.kt
package com.joyy.android_project

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val homeViewModel = ViewModelProvider(
            this,
            ViewModelProvider.NewInstanceFactory()
        ).get(HomeViewModel::class.java)
    }

}